### PR TITLE
Allow inherited classes to create Router and HttpServerOptions

### PR DESCRIPTION
**Pull request by bhiemenz**

### CODE REVIEW

File: georocket-server/src/main/java/io/georocket/GeoRocket.java [hunk 1]
```diff
@@ -343,22 +343,31 @@ private void onDelete(RoutingContext context) {
...
+    Router router = createRouter();
+    HttpServerOptions serverOptions = createHttpServerOptions();
+    HttpServer server = vertx.createHttpServer(serverOptions);
+
+    ObservableFuture<HttpServer> observable = RxHelper.observableFuture();
+    server.requestHandler(router::accept).listen(port, observable.toHandler());
+    return observable;
+  }
+
+  protected Router createRouter() {
```
Comment (michel-kraemer) [inline georocket-server/src/main/java/io/georocket/GeoRocket.java]: Protected methods should have javadoc

File: georocket-server/src/main/java/io/georocket/GeoRocket.java [hunk 2]
```diff
@@ -343,22 +343,31 @@ private void onDelete(RoutingContext context) {
...
     Router router = Router.router(vertx);
     router.get("/store/*").handler(this::onGet);
     router.post("/store/*").handler(this::onPost);
     router.delete("/store/*").handler(this::onDelete);
-    
+    return router;
+  }
+
+  protected HttpServerOptions createHttpServerOptions() {
```
Comment (michel-kraemer) [inline georocket-server/src/main/java/io/georocket/GeoRocket.java]: Protected methods should have javadoc

**Comment by michel-kraemer** [comment]
Why do you need 4 commits for only one change. Try to amend commits in the future or squash branches. (I can do it now when I merge the PR, so it can stay like this for the moment)

**Comment by bhiemenz** [comment]
I already pushed these changes, so --amend was not an option. I will use branches next time.

**Comment by michel-kraemer** [comment]
Merged through 52e9b35

**Pull request closed without merging**